File: LibraryApp/Entities/Assets/Tags/AssetTag.cs
namespace LibraryApp.Entities.Assets.Tags
{
    public class AssetTag
    {
        public int Id { get; set; }
        public int TagId { get; set; }
        public Tag Tag { get; set; }
        public int AssetId { get; set; }
        public Asset Asset { get; set; }
    }
}

File: LibraryApp/ClientApp/src/app/models/assetTag.ts
import { TagPlaceholder } from "@angular/compiler/src/i18n/i18n_ast";
import { Asset } from "./asset";
import { Tag } from "./tag";

export interface AssetTag {
    id: string;
    tagId: string;
    tag: Tag
    assetId: string;
    asset: Asset;
}